ext/SerializationExt.jl

```diff
@@ -0,0 +1,83 @@
+module SerializationExt
+export serialize, deserialize
+using LRUCache
+using Serialization
+
+# Serialization of large LRUs causes a stack overflow error, so we 
+# create a custom serializer that represents LinkedNodes as Ints
+function Serialization.serialize(s::AbstractSerializer, lru::LRU{K, V}) where {K, V}
+    # Create a mapping from memory address to id
+    node_map = Dict{Ptr, Int}()
```

Comment:
Rather than using Ptr's, I think it is simpler (and maybe more robust) to use an `IdDict`:
```suggestion
    # Create a mapping from object to id. Here we use `IdDict` to use object identity as the hash.
    node_map = IdDict{LinkedNode{K}, Int}()
```
then this can be indexed just as `node_map[node] = id` rather than using `pointer_from_objref`.

Reply:
This is great, thanks